gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java

```diff
@@ -0,0 +1,309 @@
+     * {@link #of(String, boolean)}.
+     */
+    public static GolangTranslator of(final String traversalSource) {
+        return of(traversalSource, false);
+    }
+
+    /**
+     * Creates the translator with the {@link DefaultTypeTranslator} passing the {@code withParameters} option to it
+     * which will handle type translation in a fashion that should typically increase cache hits and reduce
+     * compilation times if enabled at the sacrifice to rewriting of the script that could reduce readability.
+     */
+    public static GolangTranslator of(final String traversalSource, final boolean withParameters) {
+        return of(traversalSource, new DefaultTypeTranslator(withParameters));
+    }
+
+    /**
+     * Creates the translator with a custom {@link TypeTranslator} instance.
+     */
+    public static GolangTranslator of(final String traversalSource, final TypeTranslator typeTranslator) {
+        return new GolangTranslator(traversalSource, typeTranslator);
+    }
+
+    @Override
+    public String getTraversalSource() {
+        return this.traversalSource;
+    }
+
+    @Override
+    public Script translate(final Bytecode bytecode) {
+        return typeTranslator.apply(traversalSource, bytecode);
+    }
+
+    @Override
+    public String getTargetLanguage() {
+        return "gremlin-go";
+    }
+
+    @Override
+    public String toString() {
+        return StringFactory.translatorString(this);
+    }
+
+    ///////
+
+    /**
+     * Performs standard type translation for the TinkerPop types to Go.
+     */
+    public static class DefaultTypeTranslator extends AbstractTypeTranslator {
+
+        public DefaultTypeTranslator(final boolean withParameters) {
+            super(withParameters);
+        }
+
+        @Override
+        protected String getNullSyntax() {
+            return "nil";
+        }
+
+        @Override
+        protected String getSyntax(final String o) {
+            return "\"" + StringEscapeUtils.escapeJava(o) + "\"";
+        }
+
+        @Override
+        protected String getSyntax(final Boolean o) {
+            return o.toString();
+        }
+        @Override
+        protected String getSyntax(final Date o) {
+            return "time.Unix(" + o.getTime() + ", 0)";
+        }
+
+        @Override
+        protected String getSyntax(final Timestamp o) {
+            return "time.Unix(" + o.getTime() + ", 0)";
+        }
+
+        @Override
+        protected String getSyntax(final UUID o) {
+            return "uuid.MustParse(\"" + o.toString() + "\")";
+        }
+
+        @Override
+        protected String getSyntax(final Lambda o) {
+            // TODO: AN-1037 Lambda support in Gremlin-Go
+            // return "gremlingo.GroovyLambda(\"" + StringEscapeUtils.escapeEcmaScript(o.getLambdaScript().trim()) + "\")";
+            throw new NotImplementedException("Lambda translation not currently supported");
```

Comment:
Does the framework surrounding this accept NotImplementedException and skip it, or is this going to cause it to crash (or at least exit) at runtime?

Reply:
NotImplementedException is a RuntimeException so it will exit.